### PR TITLE
add new previewPlanID

### PR DESCRIPTION
**Pull request by voigt**
**Description**

Changes proposed in this pull request:

- introduce a new Preview Plan
- handle new Preview env var

**Related issue(s)**

- [#1686](https://github.tools.sap/kyma/backlog/issues/1686)

### CODE REVIEW

File: components/kyma-environment-broker/internal/runtimeversion/runtimeversion.go [hunk 1]
```diff
@@ -1,21 +1,36 @@
...
+		defaultPreviewVersion: previewVersion,
+		accountMapping:        accountMapping,
 	}
 }
 
 func (rvc *RuntimeVersionConfigurator) ForProvisioning(op internal.ProvisioningOperation) (*internal.RuntimeVersionData, error) {
+
 	pp := op.ProvisioningParameters
+
+	// TODO: Clarify: what about FromAccountMapping?
```
Comment (voigt) [inline components/kyma-environment-broker/internal/runtimeversion/runtimeversion.go:26]: @piotrmiskiewicz Do we need to take care of FromAccountMapping?

Comment (piotrmiskiewicz) [reply components/kyma-environment-broker/internal/runtimeversion/runtimeversion.go:26]: let's implement it

Comment (piotrmiskiewicz) [reply components/kyma-environment-broker/internal/runtimeversion/runtimeversion.go:26]: please remember about the order. The higher priority has a value provided in provisioning parameters, then the mapping, at the end - default value (from an envionment value)

File: docs/kyma-environment-broker/03-08-kyma-versions.md [hunk 2]
```diff
@@ -45,7 +46,8 @@ You can also specify a Kyma version using the **kymaVersion** provisioning param
        },
        \"parameters\": {
            \"name\": \"$NAME\",
-           \"kymaVersion\": \"$VERSION"
+           \"kymaVersion\": \"$VERSION",
+           \"kymaPreviewVersion\": \"$PREVIEW_VERSION"
```
Comment (piotrmiskiewicz) [inline docs/kyma-environment-broker/03-08-kyma-versions.md:50]: the payload will be the same - do note change it here. We need to add a note that for preview plan we have separate default kyma version.

File: docs/kyma-environment-broker/03-08-kyma-versions.md [hunk 3]
```diff
@@ -52,3 +53,5 @@ You can also specify a Kyma version using the **kymaVersion** provisioning param
 
 The **kymaVersion** provisioning parameter overrides the default settings.
 To enable this feature, set the **APP_ENABLE_ON_DEMAND_VERSION** environment variable to `true`.
+
+**Note:** If a Preview Plan is set, `kymaPreviewVersion` can be set to provision a preview version of Kyma. Alternatively `APP_KYMA_PREVIEW_VERSION` can be set as an environment variable.
```
Comment (piotrmiskiewicz) [inline docs/kyma-environment-broker/03-08-kyma-versions.md:57]: maybe something like this: "The default Kyma version for preview plan is a separate setting (APP_KYMA_PREVIEW_VERSION environment variable)"

**Review by piotrmiskiewicz — DISMISSED** [review]

**Review by klaudiagrz — DISMISSED** [review]

**Review by piotrmiskiewicz — APPROVED** [review]

**Review by PK85 — APPROVED** [review]

**Review by klaudiagrz — APPROVED** [review]

**Pull request merged**